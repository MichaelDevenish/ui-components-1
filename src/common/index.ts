export { Props } from './types'
export { getColor } from './legacy'
