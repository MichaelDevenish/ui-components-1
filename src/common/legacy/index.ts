export { getColor } from './colors/getColor'
