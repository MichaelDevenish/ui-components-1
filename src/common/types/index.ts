/**
 * For Examples, see:
 * https://github.com/Microsoft/TypeScript/blob/master/src/compiler/types.ts
 * https://github.com/alm-tools/alm/blob/master/src/common/types.ts
 */
import { Size as _s, AllSizes } from './size'

namespace Props {
  export import Size = _s
}

export {
  Props,
  AllSizes
}
