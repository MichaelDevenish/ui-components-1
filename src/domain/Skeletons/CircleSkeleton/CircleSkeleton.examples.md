#### Circle Skeleton

```jsx
<div>
  <CircleSkeleton
    showSkeleton={true}
    size='xlarge'
  />
  <CircleSkeleton
    showSkeleton={true}
    size='large'
  />
  <CircleSkeleton
    showSkeleton={true}
    size='medium'
  />
  <CircleSkeleton
    showSkeleton={true}
    size='small'
  />
</div>
```
