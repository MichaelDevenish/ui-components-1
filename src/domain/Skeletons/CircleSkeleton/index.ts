export {
  ICircleSkeletonComponentProps,
  CircleSkeleton
} from './CircleSkeleton'
