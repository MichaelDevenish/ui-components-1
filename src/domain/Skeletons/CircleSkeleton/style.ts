import styled, { css } from 'styled-components'
import { Props } from '../../../common'
import { SkeletonWrapper } from '../style'

export interface ICircleSkeletonWrapperProps {
  size?: Props.AvatarSize
}

export const CircleSkeletonWrapper = styled(SkeletonWrapper)`
  border-radius: 50%;
  line-height: 1rem;

  ${(props: ICircleSkeletonWrapperProps) => {
    switch (props.size) {
      case Props.AvatarSize.Small:
        return css`
          height: 30px;
          width: 30px;
        `
      case Props.AvatarSize.Medium:
        return css`
          height: 40px;
          width: 40px;
        `
      case Props.AvatarSize.Large:
        return css`
          height: 72px;
          width: 72px;
        `
      case Props.AvatarSize.XLarge:
        return css`
          height: 120px;
          width: 120px;
        `
    }
  }}
`
