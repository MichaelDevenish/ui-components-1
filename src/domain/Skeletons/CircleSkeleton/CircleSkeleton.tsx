import React from 'react'
import { CircleSkeletonWrapper } from './style'

export interface ICircleSkeletonOptions {
  /** If true, will display the skeleton */
  showSkeleton: boolean,
  /** Circle size of the skeleton (only applies if `shape` is set to `circle`) */
  size?: 'small' | 'medium' | 'large' | 'xlarge'
}

export interface ICircleSkeletonComponentProps {
  /** Skeleton setting */
  skeletonOptions?: ICircleSkeletonOptions
  /** Additional class names for the parent container */
  className?: string
}

class CircleSkeleton extends React.Component<ICircleSkeletonComponentProps> {
  public static defaultProps: Partial<ICircleSkeletonComponentProps> = {
    skeletonOptions: {
      showSkeleton: false,
      size: 'large'
    }
  }

  public render (): JSX.Element {
    const {
      showSkeleton = false,
      size = 'large',
    } = this.props.skeletonOptions!

    const {
      children,
      className
    } = this.props

    if (!showSkeleton) {
      return (
        <React.Fragment>
          {children}
        </React.Fragment>
      )
    }

    return (
      <CircleSkeletonWrapper
        size={size}
        className={className}
      />
    )
  }
}

export {
  CircleSkeleton
}
