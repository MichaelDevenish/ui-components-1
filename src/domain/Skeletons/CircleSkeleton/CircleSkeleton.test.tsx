import React from 'react'
import { mount } from 'enzyme'
import { CircleSkeleton } from './CircleSkeleton'
import { Props } from '../../../common'

describe('<CircleSkeleton />', () => {
  it('should render a circle skeleton', () => {
    const wrapper = mount(
      <CircleSkeleton
        showSkeleton={true}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })

  it('should render a circle skeleton with set size', () => {
    const wrapper = mount(
      <CircleSkeleton
        showSkeleton={true}
        size={Props.AvatarSize.Small}
      />
    )

    expect(wrapper).toMatchSnapshot()
  })
})
