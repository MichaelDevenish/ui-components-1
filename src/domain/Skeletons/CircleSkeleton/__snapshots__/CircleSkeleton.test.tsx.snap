// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CircleSkeleton /> should render a circle skeleton 1`] = `
.c1 {
  -webkit-animation: hNBTQs 1.2s ease-in-out infinite;
  animation: hNBTQs 1.2s ease-in-out infinite;
  background-color: #F5F9FC;
  background-image: linear-gradient(90deg,#F5F9FC,#E1E6EB,#F5F9FC);
  background-repeat: no-repeat;
  background-size: 200px 100%;
  border: 1px solid #F5F9FC;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c0 {
  border-radius: 50%;
  line-height: 1rem;
  height: 72px;
  width: 72px;
}

<CircleSkeleton
  showSkeleton={true}
  size="large"
>
  <Styled(styled.span)
    size="large"
  >
    <styled.span
      className="c0"
      size="large"
    >
      <span
        className="c0 c1"
        size="large"
      />
    </styled.span>
  </Styled(styled.span)>
</CircleSkeleton>
`;

exports[`<CircleSkeleton /> should render a circle skeleton with set size 1`] = `
.c1 {
  -webkit-animation: hNBTQs 1.2s ease-in-out infinite;
  animation: hNBTQs 1.2s ease-in-out infinite;
  background-color: #F5F9FC;
  background-image: linear-gradient(90deg,#F5F9FC,#E1E6EB,#F5F9FC);
  background-repeat: no-repeat;
  background-size: 200px 100%;
  border: 1px solid #F5F9FC;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c0 {
  border-radius: 50%;
  line-height: 1rem;
  height: 30px;
  width: 30px;
}

<CircleSkeleton
  showSkeleton={true}
  size="small"
>
  <Styled(styled.span)
    size="small"
  >
    <styled.span
      className="c0"
      size="small"
    >
      <span
        className="c0 c1"
        size="small"
      />
    </styled.span>
  </Styled(styled.span)>
</CircleSkeleton>
`;
